tests/unit/test_endpoint_provider.py

```diff
@@ -278,36 +278,32 @@ def test_provider_unit_sets_fqdn_if_not_address_on_relation_joined(self):
         self.assertEqual(data["prometheus_scrape_unit_address"], "some.host")
         self.assertIn("prometheus_scrape_unit_name", data)
 
-    @patch_network_get()
-    def test_provider_supports_multiple_jobs(self):
+    def _get_data(self, target):
```

Comment:
Can we generalize this a function?
Seems odd to me that"
- `provider` is hard-coded
- a non-test function is making assertions
- this specifically addresses app data (cf. https://github.com/canonical/operator/issues/834)

On one hand it reduces duplication but on the other hand the tests become less straightforward.